From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Wed, 28 Oct 2020 17:31:21 +0800
Subject: cursor-tracker: Get the GDK (X11) pointer position in floats,
 not ints

This makes X11 consistent with Wayland which already uses floats.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1519
---
 src/backends/meta-cursor-tracker.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/backends/meta-cursor-tracker.c b/src/backends/meta-cursor-tracker.c
index c6afd75..caac582 100644
--- a/src/backends/meta-cursor-tracker.c
+++ b/src/backends/meta-cursor-tracker.c
@@ -471,13 +471,13 @@ get_pointer_position_gdk (graphene_point_t *point,
   GdkSeat *gseat;
   GdkDevice *gdevice;
   GdkScreen *gscreen;
-  int x, y;
+  double x, y;
 
   gseat = gdk_display_get_default_seat (gdk_display_get_default ());
   gdevice = gdk_seat_get_pointer (gseat);
 
   /* Even if point is NULL we need this to get gscreen */
-  gdk_device_get_position (gdevice, &gscreen, &x, &y);
+  gdk_device_get_position_double (gdevice, &gscreen, &x, &y);
 
   if (point)
     {
